### PR TITLE
Add k8s-snap track promotion script

### PR DESCRIPTION
**Pull request by bschimke95**
Automatically promote snap revisions to next risk level after a certain amount of time.
A revision will not be promoted to stable unless there is already a stable release for this track.

Example promotion run: https://github.com/canonical/canonical-kubernetes-release-ci/actions/runs/9206894007/job/25325757212

### CODE REVIEW

**Review by louiseschmidtgen — COMMENTED** [review]
Great work on this Ben! I'm absolutely loving this automation. Ein paar comments.

File: scripts/promote-tracks.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        else:
+            released_at_date = None
+
+        if (
+            released_at_date
+            and (now - released_at_date).days > DAYS_TO_STAY_IN_RISK[risk]
+            and channels.get(f"{track}/{risk}", {}).get("revision")
+            != channels.get(f"{track}/{next_risk}", {}).get("revision")
+        ):
+            if next_risk == "stable" and not f"{track}/stable" in channels.keys():
```
[the remark this answers is not in the document]
Comment (bschimke95) [reply scripts/promote-tracks.py:89]: The dict is created here: https://github.com/canonical/canonical-kubernetes-release-ci/pull/1/files/f84aeb326c7810e6a19f0539a65596739d5a9f9b#diff-d3b0e0555affa3eb12c61bef8b514fe1827bb8f04fe753700e5a69f7c0946d88R59

The channel-map contains all channel details as documented in the API: https://dashboard.snapcraft.io/docs/reference/v2/en/snaps.html#api-snap-channel-map

[the remark this answers is not in the document]
Comment (louiseschmidtgen) [reply scripts/promote-tracks.py:89]: Ah perfect thank you for clarifying!

**Review by louiseschmidtgen — APPROVED** [review]
LGTM, thanks for providing the example run!

**Pull request merged**